compiler/modules/front.syntax/parse.bal

```diff
@@ -207,9 +207,19 @@ function parseError(Tokenizer tok, string? detail = ()) returns err:Syntax {
 }
 
 public function defnLocation(ModuleLevelDefn defn) returns d:Location {
-    return d:location(defn.part.file, defn.namePos);
+    return qNameLocationInDefn(defn, defn.namePos);
 }
 
-public function locationInDefn(ModuleLevelDefn defn, Position pos) returns d:Location {
-    return d:location(defn.part.file, pos);
+public function qNameLocationInDefn(ModuleLevelDefn defn, Position qnamePos) returns d:Location {
+    Position endPos = defn.part.file.qNameEndPos(qnamePos);
+    return locationInDefn(defn, qnamePos, endPos);
+}
+
+public function locationInDefn(ModuleLevelDefn defn, Position startPos, Position? endPos=()) returns d:Location {
```

Comment:
I would have thought this should take a Range.

Reply:
fixed